File: server/src/main/java/message/scmessage/roommanagement/ScLeaveRoom.java
package message.scmessage.roommanagement;

import entity.user.User;
import message.scmessage.ScMessage;

public class ScLeaveRoom extends ScMessage {
    public int statusCode;
    public String userID;

    public ScLeaveRoom(int statusCode) {
        this.statusCode = statusCode;
    }

    public ScLeaveRoom(int statusCode, User user) {
        this.statusCode = statusCode;
        this.userID = user.getUserID().toString();
    }
}

File: Client/Fruitster/Assets/Script/NetConnection/Messages/ScMessages/Room/ScLeaveRoom.cs
using System;
using System.Net;
using UnityEngine;
using UnityEngine.SceneManagement;

namespace Script.Messages.ScMessages
{
    [Serializable]
    public class ScLeaveRoom : ScMessage
    {
        public string userID;
        public HttpStatusCode statusCode;
        public override void OnMessage(Session session)
        {
            if (statusCode == HttpStatusCode.OK)
            {
                if (userID == UserProperties.MainPlayer.userID)
                {
                    UserProperties.UserRoom = null;
                    SingletonDontDestroy.Instance.DoAction(() => SceneManager.LoadScene("AvatarRoomSelectScene"));

                }
                else
                {
                    foreach (var user in UserProperties.UserRoom.Players.ToArray())
                    {
                        if (user.userID==userID)
                        {
                            UserProperties.UserRoom.Players.Remove(user);
                            break;
                        }
                    }
                    SingletonDontDestroy.Instance.DoAction(() => RoomSceneScript.Instance.RefreshRoom());
                }
            }
            else
            {
                Debug.Log(statusCode);
            }
        }
    }
}